classes/class-dintero-checkout-embedded.php

```diff
@@ -161,5 +162,51 @@ public function update_dintero_checkout_session() {
 
 		Dintero()->api->update_checkout_session( $session_id );
 	}
+
+	/**
+	 * Maybe set the selected pickup point in the shipping method.
+	 *
+	 * @param array $packages The shipping packages.
+	 * @return array
+	 */
+	public function maybe_set_selected_pickup_point( $packages ) {
+		$merchant_reference = WC()->session->get( 'dintero_merchant_reference' );
+		$chosen_shipping    = get_transient( "dintero_shipping_data_{$merchant_reference}" );
+
+		if ( empty( $chosen_shipping ) ) {
+			return $packages;
+		}
+
+		$is_pickup = 'pick_up' === $chosen_shipping['delivery_method'];
+		if ( ! $is_pickup ) {
+			return $packages;
+		}
+
+		// Loop each package.
+		foreach ( $packages as $package ) {
+			/**
+			 * Shipping rate object.
+			 *
+			 * @var WC_Shipping_Rate $rate
+			 */
+			foreach ( $package['rates'] as $rate ) {
+				if ( $chosen_shipping['id'] !== $rate->get_id() ) {
+					continue;
+				}
+
+				// Find the shipping rate that has the following operator product ID.
+				$id           = $chosen_shipping['operator_product_id'];
+				$pickup_point = Dintero()->pickup_points()->get_pickup_point_from_rate_by_id( $rate, $id );
+
+				if ( empty( $pickup_point ) ) {
+					continue;
+				}
+
+				Dintero()->pickup_points()->save_selected_pickup_point_to_rate( $rate, $pickup_point );
```

Comment:
Could we possibly break here? Since i believe this can only ever be valid for a single shipping method right?